### PR TITLE
DataWriter qos methods

### PR DESCRIPTION
**Pull request by laura-eprosima**
* Change DataWriterQos to add PublisherAttributes necessary data
* Implementation of PIM set_default_datawriter_qos method
* New unit test for PIM methods
* Implementation of PSM default_datawriter_qos methods
* New unit test for PSM methods

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1818/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1565/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1812/)

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 1]
```diff
@@ -919,7 +919,7 @@ public:
          * @param data data to copy in the newly created object \
          */                                                                            \
         RTPS_DllAPI TClassName(                                                            \
-            const TClassName& data) = default;                                         \
+            const TClassName &data) = default;                                         \
```
Comment (richiware) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:922]: Coding style.

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 2]
```diff
@@ -930,7 +930,7 @@ public:
          * @param data data to copy in the newly created object \
          */                                                                            \
         RTPS_DllAPI TClassName(                                                            \
-            const collection_type& data)                                               \
+            const collection_type &data)                                               \
```
Comment (richiware) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:933]: Coding style.

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 3]
```diff
@@ -947,7 +947,7 @@ public:
          * @return reference to the current object. \
          */                                                                            \
         TClassName& operator =(                                                            \
-            const TClassName& b) = default;                                            \
+            const TClassName &b) = default;                                            \
```
Comment (richiware) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:950]: Coding style.

File: include/fastdds/dds/publisher/qos/DataWriterQos.hpp [hunk 4]
```diff
@@ -20,113 +20,768 @@
...
+               (this->publish_mode_ == b.publish_mode()) &&
+               (this->representation_ == b.representation()) &&
+               (this->properties_ == b.properties()) &&
+               (this->reliable_writer_data_ == b.reliable_writer_data()) &&
+               (this->endpoint_data_ == b.endpoint_data()) &&
+               (this->writer_resources_ == b.writer_resources()) &&
+               (this->throughput_controller_ == b.throughput_controller());
+    }
+
+    RTPS_DllAPI bool check_qos() const;
```
Comment (richiware) [inline include/fastdds/dds/publisher/qos/DataWriterQos.hpp:97]: Get out this function from the class.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1823/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1570/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1817/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1876/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1617/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1869/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1880/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1620/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1873/)

File: src/cpp/fastdds/publisher/PublisherImpl.cpp [hunk 5]
```diff
@@ -373,12 +373,14 @@ ReturnCode_t PublisherImpl::set_default_datawriter_qos(
 {
     if (&qos == &DATAWRITER_QOS_DEFAULT)
     {
+        std::cout << "Default qos" << std::endl;
```
Comment (IkerLuengo) [inline src/cpp/fastdds/publisher/PublisherImpl.cpp:376]: Convert to LOG or remove

File: src/cpp/fastdds/publisher/PublisherImpl.cpp [hunk 6]
```diff
@@ -373,12 +373,14 @@ ReturnCode_t PublisherImpl::set_default_datawriter_qos(
...
         DataWriterImpl::set_qos(default_datawriter_qos_, DATAWRITER_QOS_DEFAULT, true);
     }
 
     ReturnCode_t ret_val = DataWriterImpl::check_qos(qos);
     if (!ret_val)
     {
+        std::cout << "Wrong check_qos" << std::endl;
```
Comment (IkerLuengo) [inline src/cpp/fastdds/publisher/PublisherImpl.cpp:383]: Convert to LOG or remove

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1886/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1626/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1879/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1890/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1629/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1883/)

**Pull request merged**